Plot column vignette (#37)
palaeoverse/rmacrostrat: vignettes/stratigraphic-column.Rmd

```{r lithologies, eval = TRUE}
# Print the lithologies contained in the San Juan Basin column
san_juan$lith
```

So our column contains a total of 25 different lithologies, including sedimentary, metamorphic, and igneous rocks. We can quickly visualize the proportion of the column made up of these different rocks, colored by their class:

```{r lith_plot, eval = TRUE}
# Extract lithologies
lith_table <- san_juan$lith[[1]]
# Plot lithologies by proportion in San Juan Basin column
ggplot(lith_table, aes(y = prop, x = reorder(name, -prop))) +
  # Plot bars
  geom_bar(stat = "identity", aes(fill = class)) +
  # Label axes
  scale_x_discrete("Lithology") +
  scale_y_continuous("Proportion of stratigraphic column") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

It seems the San Juan Basin is dominated by sedimentary rocks. Perhaps this is not surprising given that we know it is famed for fossils and energy resources!
Speaking of these, let's take a look at the column's economic attributes:

```{r econ, eval = TRUE}
# Print a summary of economic attributes of the San Juan Basin column
san_juan$econ
```

> Here we can see that not only does the San Juan Basin contain coal, oil, and natural gas, but also some uranium ore and aquifers.

Reviewer: ```suggestion
Here we can see that not only does the San Juan Basin contain coal, oil, and natural gas, but it also has some uranium ore and aquifers.
```

sum(sj_fossils$pbdb_occs)
```

So over 2000 fossils are known from this single basin. We can also visualize the temporal distribution of these fossil collections, using the midpoint of their age ranges:

```{r fossil_plot, eval = TRUE}
# Add midpoint age for plotting
sj_fossils$m_age <- (sj_fossils$b_age + sj_fossils$t_age) / 2

# Plot of fossil occurrence counts from the San Juan Basin through time
ggplot(sj_fossils, aes(x = m_age)) + 
  # Histogram with bars 5 million years wide
  geom_histogram(binwidth = 5, boundary = 0) +
  # Label y-axis
  scale_y_continuous("Number of fossil collections") +
  # Label x-axis and reverse direction
  scale_x_reverse("Time (Ma)", breaks = c(250, 200, 150, 100, 50)) +
  # Theming 
  theme_bw() +
  # Add geological timescale
  coord_geo()
```

From this plot, we can see that the San Juan Basin fossils range in age from the Permian to the Paleogene, but we have the highest concentration of fossil collections around the K-Pg boundary.
Now that we have explored the data in the [Macrostrat](https://macrostrat.org) database on the San Juan Basin, it is time to plot our stratigraphic column.

# Plotting the stratigraphic column

> To plot the stratigraphic column, we will need to obtain data for each lithological unit contained within the San Juan Basin column. We will do this using another `rmacrostrat` function, namely `get_units`, by referencing the `column_id` for the San Juan Basin. To keep our column contained, we will limit it to geological units which are Cretaceous in age:

Reviewer: ```suggestion
To plot the stratigraphic column, we will need to obtain data for each lithological unit contained within the San Juan Basin column. We will do this using another `rmacrostrat` function, `get_units`, and referencing the `column_id` for the San Juan Basin. To keep our column plot contained, we will limit it to geological units which are Cretaceous in age:
```

```{r source_units, eval = TRUE}
# Using the column ID, retrieve the units in the San Juan Basin column
san_juan_units <- get_units(column_id = 489, interval_name = "Cretaceous")
# See the first few rows

> head(san_juan_units)

Reviewer 2: This seems like a good approach to me.

```{r column_a, eval = TRUE, fig.height = 8}
# Plot stratigraphic column
ggplot(san_juan_units, aes(ymin = b_age, ymax = t_age, xmin = 0, xmax = 1)) +
  # Plot units
  geom_rect(alpha = 0.5, color = "black", linetype = 2) +
  # Reverse direction of y-axis
  scale_y_reverse(limits = c(145, 66), n.breaks = 10, name = "Time (Ma)") +
  # Theming
  theme_classic() +
  theme(legend.position = "none",
        axis.line.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
```

We can already see something that roughly resembles a stratigraphic column. However, there is a lot we can do to improve the aesthetics of our plot. For example, the column named `color` in our dataframe specifies the hexadecimal color corresponding to the dominant lithology of the unit. We can use this to color-code the units by lithology.

```{r column_b, eval = TRUE, fig.height = 8}
# Plot stratigraphic column
ggplot(san_juan_units, aes(ymin = b_age, ymax = t_age, xmin = 0, xmax = 1)) +

>   # Plot units

Reviewer 2: ```suggestion
  # Plot units, filling by the `color` column
```

Not great language, but thought it wise to differentiate from the code above

        axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
```

Great! Now let's add labels indicating the names of the different units.

```{r column_c, eval = TRUE, fig.height = 8}
# Add midpoint age for plotting
san_juan_units$m_age <- (san_juan_units$b_age + san_juan_units$t_age) / 2

# Plot stratigraphic column
ggplot(san_juan_units, aes(ymin = b_age, ymax = t_age, xmin = 0, xmax = 1)) +
  # Plot units
  geom_rect(alpha = 0.5, fill = san_juan_units$color, color = "black",
            linetype = 2) +
  # Add text labels
  geom_text_repel(aes(x = 1, y = m_age, label = unit_name),
                  size = 4, hjust = 0, force = 2,
                  min.segment.length = 0, direction = "y",
                  nudge_x = rep_len(x = c(2, 3), length.out = 17)) +
  # Reverse direction of y-axis
  scale_y_reverse(limits = c(145, 66), n.breaks = 10, name = "Time (Ma)") +
  # Theming
  theme_classic() +
  theme(legend.position = "none",
        axis.line.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
```

> And finally, we can add an additional facet on the y-axis indicating the different stages of the Cretaceous, using the R package `deeptime`:

Reviewer: "Facet" has a specific meaning in ggplot2-speak.
```suggestion
And finally, we can add a column along the y-axis indicating the different stages of the Cretaceous, using the R package `deeptime`:
```

```{r column_d, eval = TRUE, fig.height = 8}
# Plot stratigraphic column
ggplot(san_juan_units, aes(ymin = b_age, ymax = t_age, xmin = 0, xmax = 1)) +
  # Plot units
  geom_rect(alpha = 0.5, fill = san_juan_units$color, color = "black",
            linetype = 2) +
  # Add text labels
  geom_text_repel(aes(x = 1, y = m_age, label = unit_name),
                  size = 3.5, hjust = 0, force = 2,
                  min.segment.length = 0, direction = "y",
                  nudge_x = rep_len(x = c(2, 3), length.out = 17)) +
  # Add additional unit rectangles
  geom_rect(aes(ymin = b_age, ymax = t_age, xmin = 0.9, xmax = 1),

>             fill = san_juan_units$color, color = "black") +

Reviewer: What's the point of these rectangles? It doesn't look bad, but it might be good to explain to the reader why we are adding these.

Reviewer 2: Agreed!

Author: In the end I removed them as I don't think they were doing the job they were intended for.
